src/sklearndf/wrapper/_wrapper.py

```diff
@@ -448,15 +449,29 @@ def _post_fit(
             self._outputs = y.columns.tolist()
 
     # noinspection PyPep8Naming
-    def _check_parameter_types(
+    def _validate_parameter_types(
         self,
-        X: pd.DataFrame,
+        X: Union[pd.Series, pd.DataFrame],
         y: Optional[Union[pd.Series, pd.DataFrame]],
         *,
         expected_columns: pd.Index = None,
-    ) -> None:
-        if not isinstance(X, pd.DataFrame):
-            raise TypeError("arg X must be a DataFrame")
+    ) -> Tuple[pd.DataFrame, Union[pd.Series, pd.DataFrame]]:
```

Comment:
Just a nitpick - I think return type of this method for `y` part should be `Union[pd.Series, pd.DataFrame, None]` as `y` argument can be a `None` and is accepted.

Reply:
Well spotted, thanks. Fixed.